packages/e2e/test/local-network/local-network.test.ts

```diff
@@ -110,4 +161,67 @@ describe('Local Network', () => {
     wallet1.shutdown();
     wallet2.shutdown();
   });
+
+  it('will receive rewards for delegated tADA', async () => {
+    // Arrange
+    const amountFromFaucet = 100_000_000_000;
+
+    const wallet1 = await getWallet({ env, logger, name: 'Sending Wallet', polling: { interval: 50 } });
+    const wallet2 = await getWallet({ env, logger, name: 'Receiving Wallet', polling: { interval: 50 } });
+
+    await firstValueFrom(wallet1.wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+    await firstValueFrom(wallet2.wallet.syncStatus.isSettled$.pipe(filter((isSettled) => isSettled)));
+
+    const [{ address: sendingAddress }] = await firstValueFrom(wallet1.wallet.addresses$);
+
+    const waitForEpoch = async (numberOfEpochsToWait: number) => {
+      const currentEpoch = await firstValueFrom(wallet1.wallet.currentEpoch$);
+      return await firstValueFrom(
+        wallet1.wallet.currentEpoch$.pipe(
+          filter(({ epochNo }) => epochNo > currentEpoch.epochNo + numberOfEpochsToWait)
+        )
+      );
+    };
+
+    logger.debug(`Address ${sendingAddress.toString()} will be funded with ${amountFromFaucet} tLovelace.`);
+    await faucetProvider.request(sendingAddress.toString(), amountFromFaucet, 1);
```

Comment:
Currently the faucet is not network agnostic since we don't have an analogous (yet) on the other networks. I think it should be fine since I don't think we can run this test on regular networks (since each epoch last 5 days), but then if we want to run this on CI we will need to also add:

```
FAUCET_PROVIDER=cardano-wallet
FAUCET_PROVIDER_PARAMS='{"baseUrl":"http://localhost:8090/v2","mnemonic":"fire method repair aware foot tray accuse brother popular olive find account sick rocket next"}'
```
In the [continuous-integration.yaml](https://github.com/input-output-hk/cardano-js-sdk/blob/ae8a3853ce794394463fd501fe95e4f9d1fadb32/.github/workflows/continuous-integration.yaml#L57) file

Reply:
Hi @AngelCastilloB , 
as far as I understood, in CI we are sure the test will run against local-network, so using the configuration you provided should not be source of problems. Am I right?

While about the _wait for epoch_ problem; if is there a way to know it the current network doesn't have an acceptable epoch length, we could selectively choose to skip or not the test...
Do you know if is there some information which can be useful to take this decision?
Thank you

Reply:
[epochLength](https://github.com/input-output-hk/cardano-js-sdk/blob/7283619e62667fe5bb672b2903b29b06776ef3eb/packages/core/src/Cardano/types/Genesis.ts#L13) * [slotLength](https://github.com/input-output-hk/cardano-js-sdk/blob/7283619e62667fe5bb672b2903b29b06776ef3eb/packages/core/src/Cardano/types/Genesis.ts#L16) * 1000 gives you the epoch duration in seconds. [NetworkInfo.genesisParameters()](https://github.com/input-output-hk/cardano-js-sdk/blob/0cd887737cc5d4f8d920405c803f05f2c47e42f2/packages/core/src/Provider/NetworkInfoProvider/types.ts#L30)